### ISSUE TITLE
THIS_FILE macro conflicts with CocoaLumberjack macro

### ISSUE THREAD

**Issue by bvirlet** [aws/aws-sdk-ios#895]
### To help us solve your problem better, please answer the following list of questions.

* What service are you using?

AWSCore

* In what version of SDK are you facing the problem?

2.6.14

* Is the issue limited to Simulators / Actual Devices?

Compilation

* Can your problem be resolved if you bump to a higher version of SDK?

No

* Is this problem related to specific iOS version?

No.
Reproduced with Xcode 9.3.

* How are you consuming the SDK? CocoaPods / Carthage / Prebuilt frameworks?

CocoaPods or direct framework.

* Can you give us steps to reproduce with a minimal, complete, and verifiable example? Please include any specific network conditions that might be required to reproduce the problem.

Yes:

- Create a project that depends on CocoaLumberjack
- Add AWSCore pod (or directly as a framework dependency)
- Notice warning: 

> DDLog.h:188:9: 'THIS_FILE' macro redefined

```
/**
 * The THIS_FILE macro gives you an NSString of the file name.
 * For simplicity and clarity, the file name does not include the full path or file extension.
 *
 * For example: DDLogWarn(@"%@: Unable to find thingy", THIS_FILE) -> @"MyViewController: Unable to find thingy"
 **/
#define THIS_FILE         (DDExtractFileNameWithoutExtension(__FILE__, NO))
```

The problem is that AWSCore copied over the CocoaLumberjack files but didn't rename all the variables. THIS_FILE should be renamed differently to not conflict with CocoaLumberjack.

**Comment by rohandubal** [comment]
Hello @bvirlet Thanks for reporting this! We will try to get this fix soon, in the meanwhile, would it possible for you to send us a PR? Thanks!

**Comment by bvirlet** [comment]
TBH I don't know the right way of fixing this!

**Comment by cbommas** [comment]
Hello @bvirlet 
We have fixed this and it will be included in the next rev of the SDK.

**Comment by kvasukib** [comment]
This has been fixed in 2.6.17. Please upgrade to the latest and let us know if it fixes the issue. Closing the issue.

**Comment by yogesh-zendrive** [comment]
This issue still exists in AWS SDK 2.6.31
